examples/breakpoint-emulate-example.c

```diff
@@ -0,0 +1,215 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Author: Mathieu Tarral (mathieu.tarral@ssi.gouv.fr)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This example sets a software breakpoint on a given symbol, and when the callback is called,
+ * it requests to emulate a given opcode, read before placing the breakpoint,
+ * which is significantly faster than singlestepping
+ *
+ * You have to specify the opcode size.
+ *
+ * For example for NtOpenFile on a Windows 7, the opcode size is 3
+ *
+ * [0x1403690dc]> pd 10
+            ;-- ntoskrnl.exe_NtOpenFile:
+            ;-- pdb.NtOpenFile:
+            0x1403690dc      4c8bdc         mov r11, rsp
+            0x1403690df      4881ec880000.  sub rsp, 0x88
+            0x1403690e6      8b8424b80000.  mov eax, dword
+            0x1403690ed      4533d2         xor r10d, r10d
+
+ * In this case, 0x4c8bdc will be the opcode to be emulated.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <signal.h>
+
+#include <libvmi/libvmi.h>
+#include <libvmi/events.h>
+
+struct cb_data {
+    char *symbol;
+    addr_t vaddr;
+    emul_insn_t emul;
+};
+
+event_response_t int3_cb(vmi_instance_t vmi, vmi_event_t *event)
+{
+    vmi = vmi;
```

Comment:
Please change this to `(void)vmi;` to silence warnings generated on clang-scan

Reply:
Done